modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py

```diff
@@ -854,24 +841,36 @@ async def stop():
         await asyncio.sleep(1)
         asyncio.get_running_loop().stop()
 
-    def signal_handler():
+
+    def signal_handler(signum, frame):
+        global mdm_agent_stopped
+
+        sig_name = signal.Signals(signum).name
+        cc.logger.debug(f"signal_handler {sig_name} ({signum}) received.")
+
+        if sig_name == "SIGUSR1":
```

Comment:
Shouldn't SIGINT and SIGTERM also be checked for here? Maybe someone gets impatient and tries to stop the MDM agent twice in a row :thinking: